Issue 21219: improve error handling for botDisableCmdF

ariyonaty opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->
Error handling for the `botDisableCmdF` function updated to return an error in case of failure. 
Update includes the use of `multierror.Error` for appending all errors and returning error if one occurs.

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

Fixes https://github.com/mattermost/mattermost-server/issues/21219

Otherwise, link the JIRA ticket.
-->

Fixes https://github.com/mattermost/mattermost-server/issues/21219

mattermod commented:
Hello @ariyonaty,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

noxer commented:
Hi @ariyonaty, thank you for your contribution. I'll run some tests and review the code.

isacikgoz reviewed (approved):
Awesome, LGTM 🎉

noxer commented on commands/bot.go:
```diff
@@ -236,22 +237,26 @@ func botEnableCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 	for i, user := range users {
 		if user == nil {
 			printer.PrintError(fmt.Sprintf("can't find user '%v'", args[i]))
+			result = multierror.Append(result, fmt.Errorf("can't find user '%v'", args[i]))
```
Please use `%q` to quote strings, the original code was lacking in that regard as well. It would be awesome if you could change it there as well.

noxer commented on commands/bot.go:
```diff
@@ -236,22 +237,26 @@ func botEnableCmdF(c client.Client, cmd *cobra.Command, args []string) error {
 		bot, _, err := c.DisableBot(user.Id)
 		if err != nil {
 			printer.PrintError(fmt.Sprintf("could not disable bot '%v'", args[i]))
+			result = multierror.Append(result, fmt.Errorf("could not disable bot '%v'", args[i]))
```
Could you please add the `err` to the message?

noxer commented on commands/bot_e2e_test.go:
```diff
@@ -284,7 +284,7 @@ func (s *MmctlE2ETestSuite) TestBotDisableCmd() {
 
 		err := botDisableCmdF(s.th.Client, &cobra.Command{}, []string{newBot.UserId})
-		s.Require().Nil(err)
+		s.Require().NotNil(err)
```
Please check for errors using `s.Require().Error(err)`, it makes the messages more readable.

noxer commented on commands/bot_e2e_test.go:
```diff
@@ -295,7 +295,7 @@ func (s *MmctlE2ETestSuite) TestBotDisableCmd() {
 
 		err := botDisableCmdF(c, &cobra.Command{}, []string{"nonexistent-bot-userid"})
-		s.Require().Nil(err)
+		s.Require().NotNil(err)
```
Same as above

noxer reviewed (changes requested):
Looks good, a few small change requests from my side.

ariyonaty commented:
@noxer - Thank you for the feedback.
Addressed the comments in commit [6daf6bd](https://github.com/mattermost/mmctl/pull/576/commits/6daf6bd525b95a8d78e8b17a77dcf7ff6021d14f).

mattermod commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

noxer reviewed (approved):
Perfect, thank you @ariyonaty :1st_place_medal:
